### ISSUE TITLE
Resolving ViewModels with viewModel in Koin module causes issue with data binding

### ISSUE THREAD

**Issue by NukeLar** [InsertKoinIO/koin#660]
viewModel declaration in Koin module provides a bug with two-way data binding

1. Create Activity with Viewmodel attached to it
`private val viewModel: StartViewModel by viewModel()`
2. Resolve viewmodel as following in Koin module: 
```
val vmModule = module {
    viewModel { StartViewModel(get(), get()) }
}
```
3. Setup databinding:
```
val binding: StartActivityBinding =
            DataBindingUtil.setContentView(this, R.layout.start_activity)

        binding.viewModel = viewModel
        binding.lifecycleOwner = this
```
4. Create MutableLiveData field in StartViewModel:
`var email: MutableLiveData<String> = _email`
5. Setup two-way databinding for this property in any view
```
<EditText
            android:layout_width="match_parent"
            android:layout_height="56dp"
            android:text="@={viewModel.email}"/>
```
6. Observe this property from activity: 
`viewModel.email.observe(this, Observer { handleEmailChange(it) })`

Expected behaviour:
Upon changing data from view, it should trigger 'handleEmailChange' fun. 
What rly happens:
Data binding works fine, LiveData property changes its value, however, activity is not notified.
If property is changed from any other source except the view (two-way data binding) activity gets notified.
I've tried to create a SingleLiveEvent and trigger it manually from viewModel class, when the observed LiveData changes, but activity still can't get the event.
Basically activity can't observe anything if it's triggered by the view.

I use
```
    implementation "org.koin:koin-android:2.0.1"
    implementation "org.koin:koin-android-viewmodel:2.0.1"
```

This is my first ticket, so please, let me know if everything is clear and if You need any additional information
